Automattic/newspack-plugin
# feat(ras-acc): minor visual fixes

### All Submissions:


### Changes proposed in this Pull Request:

This PR, along with https://github.com/Automattic/newspack-blocks/pull/1747, fix a handful of spacing and sizing issues in the Newspack UI components, and updates the UI font to Inter so it's consistent across all devices. 

Some specific fixes include:

* Fixing an alignment issue with the coupon field and adding error styles
* Fixing button spacing issues
* Updating the select2 colours (state/province dropdown) to match the Newspack UI theme.
* Removing the `!important` styles from `label`s and adding a WC-specific override instead.
* Fixing the transaction table font sizing
* Fixing spacing around the 'shipping address' toggle.

### How to test the changes in this Pull Request:

I've broken out the testing steps by each thing to review, but you could also just apply the PRs and run `npm run build`, and check each of these sections to make sure they look okay. This PR is also available on [the components test site](https://newspack-components-ui-demo.newspackstaging.com/).

#### Testing shipping toggle, select2 colours, and general spacing fixes:

1. Create a simple, non-virtual product so it can be shipped.
2. Add a checkout button block to a page and assign it to your non-virtual product.
3. Run through the checkout; confirm that the select2 now uses the same colour as the buttons, and not blue:

![image](https://github.com/Automattic/newspack-plugin/assets/177561/b8e05db9-8e2c-4344-9ccb-ad0eb91a96ab)

4. At the bottom of the 'Billing' section, note the appearance of the 'Different shipping address' toggle. Turn it on and fill out your shipping address and click 'Continue'.

![image](https://github.com/Automattic/newspack-plugin/assets/177561/4dd61f92-1921-4d1f-ac54-54a3835afe5b)

![image](https://github.com/Automattic/newspack-plugin/assets/177561/906c7df8-6540-4113-81cd-0cd1b5cf4b50)

5. On the next screen, note the spacing, especially for the Billing, Shipping, Payment headers. Also note the inconsistent font sizing in the Transaction Details table:

![image](https://github.com/Automattic/newspack-plugin/assets/177561/6f906367-a5a7-427b-ac4d-79cc9a3e5ed9)

![image](https://github.com/Automattic/newspack-plugin/assets/177561/01675ca5-6c37-413d-af41-588763706cc1)

6. Apply this PR and [the block PR](https://github.com/Automattic/newspack-blocks/pull/1747), and run `npm run build` for each. 
7. Repeat the above testing steps and confirm that the spacing and sizing seems more consistent in each section.

![image](https://github.com/Automattic/newspack-plugin/assets/177561/14fcc609-d4ec-44a6-ae25-171698fa1f18)

![image](https://github.com/Automattic/newspack-plugin/assets/177561/8a19a72b-5b18-4869-b6e4-99eb1093d27a)

![image](https://github.com/Automattic/newspack-plugin/assets/177561/a21813a1-7be2-4e2a-a368-e629d71662fc)

![image](https://github.com/Automattic/newspack-plugin/assets/177561/43d31f3e-9fa1-41f3-b5c1-1cbf099a3035)

#### Testing the coupon changes:

1. Navigate to WP Admin > WooCommerce > Settings, and enable coupons.
9. Create at least one coupon code under WP Admin > WooCommerce > Marketing.
10. Set up a product (subscription or simple), and add a checkout block button that is linked to it.
11. Run through the checkout process and note its appearance:

![image](https://github.com/Automattic/newspack-plugin/assets/177561/db91f666-2531-4820-97be-4c3d7997b154)

12. Apply this PR and [the corresponding block one](https://github.com/Automattic/newspack-blocks/pull/1747), and run `npm run build` for each.
13. Run through the checkout again; note the coupon form's appearance is closer to the mockups (no button alignment issue; better spacing). 
14. Try adding a fake coupon code or submitting the form without a code, and confirm you get an error message:

![image](https://github.com/Automattic/newspack-plugin/assets/177561/1f2eb3ce-b339-438e-91ab-22772bcf2c13)

15. Add your real coupon code and confirm it's successfully applied, and the error styling goes away:

![image](https://github.com/Automattic/newspack-plugin/assets/177561/5b88b139-3225-4d57-b958-ef433ed6af11)

#### Testing button spacing issues

1. Set up a variable product, and add a checkout block button to a page that loads it.
2. In an incognito window (so you have to sign up), run through the checkout and note the button spacing -- specifically:
* Note the product variation buttons -- too much space above. 
* Check the spacing of buttons in the sign up form - there's too much on the bottom.

![image](https://github.com/Automattic/newspack-plugin/assets/177561/d9823d1a-b537-4c9c-8308-5dbb5af04efa)

![image](https://github.com/Automattic/newspack-plugin/assets/177561/03b8f40e-3302-48d0-b3f0-608459a4ea49)

3. Apply this PR to the plugin & the associated block one, and run `npm run build`.
4. Repeat the checkout, and check the button spacing:

![image](https://github.com/Automattic/newspack-plugin/assets/177561/c3c5673c-4025-4615-a050-d0705ebd6d9c)

![image](https://github.com/Automattic/newspack-plugin/assets/177561/75e3c06f-8434-46eb-8662-c6b4a1985c1c)

### Other information:
